client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -34,35 +35,69 @@ const OnboardingEdit: FunctionComponent< BlockEditProps< Attributes > > = () =>
 			{ isCreatingSite && (
 				<Redirect push={ replaceHistory ? undefined : true } to={ makePath( Step.CreateSite ) } />
 			) }
-			<Switch>
-				<Route exact path={ makePath( Step.IntentGathering ) }>
-					<AcquireIntent />
-				</Route>
+			<Route exact path={ makePath( Step.IntentGathering ) }>
+				{ ( { match } ) => (
+					<CSSTransition
+						in={ match !== null }
+						unmountOnExit
+						classNames="gutenboarding-page"
+						timeout={ 240 }
+					>
+						<AcquireIntent />
+					</CSSTransition>
+				) }
+			</Route>
 
+			{ ! siteVertical && (
 				<Route path={ makePath( Step.DesignSelection ) }>
-					{ ! siteVertical ? (
-						<Redirect to={ makePath( Step.IntentGathering ) } />
-					) : (
-						<DesignSelector />
-					) }
+					<Redirect to={ makePath( Step.IntentGathering ) } />
 				</Route>
+			) }
+
+			<Route path={ makePath( Step.DesignSelection ) }>
+				{ ( { match } ) => (
+					<CSSTransition
+						in={ match !== null }
+						unmountOnExit
+						classNames="gutenboarding-page"
+						timeout={ 250 }
+					>
+						<DesignSelector />
+					</CSSTransition>
+				) }
+			</Route>
 
+			{ ! selectedDesign && ! isEnabled( 'gutenboarding/style-preview' ) && (
 				<Route path={ makePath( Step.Style ) }>
-					{ // Disable reason: Leave me alone, my nested ternaries are amazing ✨
-					// eslint-disable-next-line no-nested-ternary
-					! selectedDesign ? (
-						<Redirect to={ makePath( Step.DesignSelection ) } />
-					) : isEnabled( 'gutenboarding/style-preview' ) ? (
-						<StylePreview />
-					) : (
-						<Redirect to={ makePath( Step.DesignSelection ) } />
-					) }
+					<Redirect to={ makePath( Step.DesignSelection ) } />
 				</Route>
+			) }
 
-				<Route path={ makePath( Step.CreateSite ) }>
-					<CreateSite />
-				</Route>
-			</Switch>
+			<Route path={ makePath( Step.Style ) }>
+				{ ( { match } ) => (
+					<CSSTransition
+						in={ match !== null }
+						unmountOnExit
+						classNames="gutenboarding-page"
+						timeout={ 250 }
+					>
+						<StylePreview />
+					</CSSTransition>
+				) }
+			</Route>
+
+			<Route path={ makePath( Step.CreateSite ) }>
+				{ ( { match } ) => (
+					<CSSTransition
+						in={ match !== null }
+						unmountOnExit
+						classNames="gutenboarding-page"
+						timeout={ 250 }
+					>
+						<CreateSite />
+					</CSSTransition>
+				) }
```

Comment:
Maybe we can avoid some of the repetition while separating a bit routing logic from styling effects. Also I like how simple this is so we should think twice before adding extra complexity.

Reply:
Great point.